src/wallet/wallet.cpp

```diff
@@ -922,6 +922,37 @@ bool CWallet::MarkReplaced(const uint256& originalHash, const uint256& newHash)
     return success;
 }
 
+void CWallet::SetDirtyState(const uint256& hash, unsigned int n, bool dirty)
+{
+    const CWalletTx* srctx = GetWalletTx(hash);
+    if (srctx) {
```

Comment:
Commit "wallet: enable avoidreuse feature"

Instead of nesting, could early return here:

```cpp
if (!srctx) return;
```

Same in `CWallet::IsDirty` below.